src/Handler/Search.hs

```diff
@@ -84,15 +120,21 @@ routeResult SearchResult{..} =
   ppkgName = PathPackageName hrPkgName
   pversion = PathVersion hrPkgVersion
 
-searchForName :: Text -> Handler [SearchResult]
-searchForName query = do
+applyPagination :: Int -> [a] -> ([a], Bool)
+applyPagination page xs =
+    let ys = take (resultsPerPage + 1) $ drop ((page - 1) * resultsPerPage) xs
```

Comment:
This will give us `resultsPerPage + 1` results per page, right? That doesn't seem right :smile:

How about this?

```haskell
let
  (ys, more) = splitAt resultsPerPage $ drop ((page - 1) * resultsPerPage) xs
in
  (ys, not (null more))
```